db/db_impl/db_impl_open.cc

```diff
@@ -136,6 +137,10 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
     result.avoid_flush_during_recovery = false;
   }
 
+  if (result.enable_sst_file_checksum && !result.sst_file_checksum) {
```

Comment:
@riversand963 Answer the previous question, if enable_sst_file_checksum but sst_file_checksum is null. When open the db, it will check the flag and the pointer. If flag is set but checksum method is not specified, the default one will be used. In table_builder and compaction_job, we still need to check both in case something like sst_file_checksum is freed unexpected.

Reply:
Is this a bug?
> something like sst_file_checksum is freed unexpected.

Reply:
@riversand963 There are two cases: 1) user does not provide the sst_file_checksum method, just enables the enable_sst_file_checksum; 2) user provide initiate the pointer but sst_file_checksum is freed unexpected before this check. For case 1, it is not a bug. For case 2, it can be a bug or crash happens before this check. But db cannot decide what case should be. So we follow the flag and reset to the default Crc32. Do you mean we should log a potential error message here?